### ISSUE TITLE
`amplify push` fails with error "no such file or directory **/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json"

### ISSUE THREAD

**Issue by duranmla** [aws-amplify/amplify-cli#9146]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v14.17.4

### Amplify CLI Version

7.5.4

### What operating system are you using?

Mac - Big Sur 11.4 (M1)

### Amplify Categories

Not applicable

### Amplify Commands

push

### Describe the bug

As soon as I `amplify push` I got:

```
Build error : ENOENT: no such file or directory, open '/Users/alexis/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json'
```

This happens when I upgrade from 7.5.1 towards this 7.5.4 version. Although the downgrade seems to be an alternative, I have also realise that the Data management on the Amplify console fails with `Error: File '/. ` when I try to update the schema and the log suggest the same error that I have in my local when using 7.5.4 CLI

```
Command failed with exit code 1: /root/.amplify/bin/amplify pull --amplify {"envName":"alexis","defaultEditor":"code","appId":"d3211m2y0rd7kd"} --providers {"awscloudformation":{"configLevel":"project","useProfile":true,"profileName":"amplify-backend-processor"}} --frontend {"frontend":"javascript","framework":"react","config":{"SourceDir":"src","DistributionDir":"build","BuildCommand":"npm run-script build","StartCommand":"npm run-script start"}} --yes
- Fetching updates to backend environment: alexis from the cloud.
- Building resource auth/userPoolGroups
- Building resource auth/solauth
- Building resource api/AdminQueries
- Building resource api/sol
2021-11-30T12:02:41.139Z - error: uncaughtException: Error: Error: File '/**/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json' was not included into executable at compilation stage. Please recompile adding it as asset or script. date=Tue Nov 30 2021 12:02:41 GMT+0000 (Coordinated Universal Time), pid=492, uid=0, gid=0, cwd=/tmp/amplify-8a58a030-c675-4cb6-8b8f-bda684b13765, execPath=/root/.amplify/bin/amplify, version=v12.22.7, argv=[/root/.amplify/bin/amplify, /snapshot/node_modules/@aws-amplify/cli/bin/amplify, pull, --amplify, {"envName":"alexis","defaultEditor":"code","appId":"d3211m2y0rd7kd"}, --providers, {"awscloudformation":{"configLevel":"project","useProfile":true,"profileName":"amplify-backend-processor"}}, --frontend, {"frontend":"javascript","framework":"react","config":{"SourceDir":"src","DistributionDir":"build","BuildCommand":"npm run-script build","StartCommand":"npm run-script start"}}, --yes], rss=432218112, heapTotal=352219136, heapUsed=296243416, external=14267977, arrayBuffers=12688133, loadavg=[0.77, 0.43, 0.17], uptime=303, trace=[column=15, file=/snapshot/node_modules/@aws-amplify/amplify-category-auth/lib/provider-utils/awscloudformation/utils/generate-user-pool-group-stack-template.js, function=generateUserPoolGroupStackTemplate, line=11, method=null, native=false, column=null, file=null, function=runMicrotasks, line=null, method=null, native=false, column=5, file=internal/process/task_queues.js, function=processTicksAndRejections, line=97, method=null, native=false], stack=[Error: Error: Error: File '/**/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json' was not included into executable at compilation stage. Please recompile adding it as asset or script.,     at generateUserPoolGroupStackTemplate (/snapshot/node_modules/@aws-amplify/amplify-category-auth/lib/provider-utils/awscloudformation/utils/generate-user-pool-group-stack-template.js:11:15),     at runMicrotasks (<anonymous>),     at processTicksAndRejections (internal/process/task_queues.js:97:5)]
⚠️ Amplify command hooks are disabled in the current execution environment.
⚠️ See https://docs.amplify.aws/cli/usage/command-hooks/ for more information.
[0mAmplify AppID found: d3211m2y0rd7kd. Amplify App name is: sol[0m
[0mBackend environment alexis found in Amplify Console app: sol[0m
[32mGraphQL schema compiled successfully.[39m

[32mEdit your schema at /tmp/amplify-8a58a030-c675-4cb6-8b8f-bda684b13765/amplify/backend/api/sol/schema.graphql or place .graphql files in a directory at /tmp/amplify-8a58a030-c675-4cb6-8b8f-bda684b13765/amplify/backend/api/sol/schema[39m
[0mSuccessfully generated models. Generated models can be found in /tmp/amplify-8a58a030-c675-4cb6-8b8f-bda684b13765/src[0m
🛑 Build error : File '/**/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json' was not included into executable at compilation stage. Please recompile adding it as asset or script.
```

#### TL;DR

In local with CLI 7.5.4 I am not able to push. When downgrade I am able to push but fails with `Failed to create models in the cloud: Modelgen job creation failed` while the Amplify console under Home section says: `Error: File '/` if I try to make a Data deploy using the console I go the same error `Error: File '/` but with more context (fails because `overrides-resource/tsconfig.json` not found)

### Expected behavior

I should be able to do any amplify command and create new environments or update the schema using CLI 7.5.4

### Reproduction steps

I am pretty sure that this must be something with my project but I don't find any guidance anywhere. In my context I do:

1. `amplify -v` with v7.5.4
2. `amplify push` gets error
3. `npm i -g @aws-amplify/cli@7.5.1`
4. `amplify push` no error but fail in the end with "Failed to create models in the cloud: Modelgen job creation failed"

### GraphQL schema(s)

<details>

```graphql
type Post
  @model
  @auth(
    rules: [
      { allow: private }
      { allow: owner, operations: [create, update, delete] }
      {
        allow: groups
        groups: ["Admins"]
        operations: [read, create, update, delete]
      }
    ]
  ) {
  id: ID!
  url: String!
  openGraphData: AWSJSON
  pitstop: Pitstop @hasOne
}

type PathwayProgression @model @auth(rules: [{ allow: private }]) {
  id: ID!
  pathwayID: ID! @index(name: "byPathway")
  profileID: ID! @index(name: "byProfile")
  pathway: Pathway! @hasOne(fields: ["pathwayID"])
  profile: Profile! @hasOne(fields: ["profileID"])
  progression: AWSJSON
}

type PitstopVariant @model @auth(rules: [{ allow: private }]) {
  id: ID!
  name: String!
  config: AWSJSON
  pitstops: [Pitstop] @hasMany
}

type Pitstop @model @auth(rules: [{ allow: private }]) {
  id: ID!
  pitstopVariantID: ID!
  pitstopVariant: PitstopVariant @belongsTo(fields: ["pitstopVariantID"])
  title: String!
  description: String
  pathwayPitstops: [PathwayPitstop] @hasMany
}

type Pathway @model @auth(rules: [{ allow: private }]) {
  id: ID!
  title: String!
  description: String
  thumbnail: String
  pathwayPitstops: [PathwayPitstop] @hasMany
}

type Profile
  @model
  @auth(
    rules: [{ allow: owner }, { allow: private, operations: [read, update, delete] }]
  ) {
  id: ID!
  companies: [Company] @manyToMany(relationName: "ProfileCompany")
  professions: [Profession] @manyToMany(relationName: "ProfileProfession")
  email: String!
}

type Company
  @model
  @auth(rules: [{ allow: private, operations: [create, read, update] }]) {
  id: ID!
  name: String!
  slug: String
  profiles: [Profile] @manyToMany(relationName: "ProfileCompany")
}

type Profession
  @model
  @auth(rules: [{ allow: private, operations: [create, read, update] }]) {
  id: ID!
  name: String!
  slug: String
  profiles: [Profile] @manyToMany(relationName: "ProfileProfession")
}

type PathwayPitstop @model @auth(rules: [{ allow: private }]) {
  id: ID!
  pathwayID: ID!
  pitstopID: ID!
  pathway: Pathway! @belongsTo(fields: ["pathwayID"])
  pitstop: Pitstop! @belongsTo(fields: ["pitstopID"])
  positionX: Int
  positionY: Int
}

type PathwayEdge @model @auth(rules: [{ allow: private }]) {
  id: ID!
  pathwayPitstopID: ID! @index(name: "byPathwayPitstop")
  parentID: ID!
  pathwayPitstop: PathwayPitstop @hasOne(fields: ["pathwayPitstopID"])
  parent: PathwayPitstop @hasOne(fields: ["parentID"])
}
```

</details>

### Log output

<details>

```
🛑 Build error : ENOENT: no such file or directory, open '/Users/alexis/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json'
⠋ Building resource api/sol2021-11-30T12:09:07.680Z - error: uncaughtException: Error: Error: ENOENT: no such file or directory, open '/Users/alexis/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json' date=Tue Nov 30 2021 13:09:07 GMT+0100 (Central European Standard Time), pid=44501, uid=501, gid=20, cwd=/Users/alexis/work@/sol, execPath=/Users/alexis/.nvm/versions/node/v14.17.4/bin/node, version=v14.17.4, argv=[/Users/alexis/.nvm/versions/node/v14.17.4/bin/node, /Users/alexis/.nvm/versions/node/v14.17.4/bin/amplify, env, checkout, alexis], rss=634634240, heapTotal=506413056, heapUsed=481301224, external=15100365, arrayBuffers=12894234, loadavg=[3.7373046875, 2.576171875, 2.1044921875], uptime=1110859, trace=[column=11, file=/Users/alexis/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-user-pool-group-stack-template.ts, function=generateUserPoolGroupStackTemplate, line=10, method=null, native=false, column=null, file=null, function=runMicrotasks, line=null, method=null, native=false, column=5, file=internal/process/task_queues.js, function=processTicksAndRejections, line=95, method=null, native=false, column=5, file=/Users/alexis/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/utils/synthesize-resources.ts, function=updateUserPoolGroups, line=259, method=null, native=false, column=7, file=/Users/alexis/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts, function=AmplifyAuthTransform.transform, line=59, method=transform, native=false, column=10, file=/Users/alexis/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-stack-template.ts, function=generateAuthStackTemplate, line=6, method=null, native=false], stack=[Error: Error: Error: ENOENT: no such file or directory, open '/Users/alexis/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json',     at generateUserPoolGroupStackTemplate (/Users/alexis/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-user-pool-group-stack-template.ts:10:11),     at runMicrotasks (<anonymous>),     at processTicksAndRejections (internal/process/task_queues.js:95:5),     at updateUserPoolGroups (/Users/alexis/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/utils/synthesize-resources.ts:259:5),     at AmplifyAuthTransform.transform (/Users/alexis/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts:59:7),     at generateAuthStackTemplate (/Users/alexis/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-stack-template.ts:6:10)]
```

</details>

### Additional information

I am constantly facing issues trying to create new environments. Our expectation is that every dev have an env to work with features and then we merge to develop. As soon as we try to create new envs we have face continuos issues with several things perhaps because we try to make the migration to GraphQL transformer v2 and the update to CLI 7+

**Comment by duranmla** [comment]
This feels somehow related to this? https://github.com/aws-amplify/amplify-cli/pull/9078/commits/19329dfe6cd63258738a1300742c22b821ba4e75 But it suppose to be solved. In that sense, I am not sure about the way to move forward.

**Comment by akshbhu** [comment]
hi @duranmla 

May I know the steps  (amplify commands ) and what categories you added when you get this error?

Have you tried the command ```amplify override``` when this error came up?

**Comment by duranmla** [comment]
To reproduce the error is what I mentioned on the description of this issue:
- `npm i -g @aws-amplify/cli` to install latest version of the cli
- try to make an update on my schema and make `amplify api gql-compile` just to ensure all is correct
- `amplify push` got error

> Even if I remove the change on my schema and try to push I still got the error as soon as I try to push with version 7.5.4, furthermore, I am not even able to `amplify env checkout <env_name>` as it will fail with same error.

Categories I have added:
![CleanShot 2021-12-01 at 11 03 26@2x](https://user-images.githubusercontent.com/1425162/144214018-c8f13f41-2eea-484c-80e0-1f0b1fb1951b.png)

**Comment by duranmla** [comment]
Before, I used to migrate everything after finding [this](https://docs.amplify.aws/cli/migration/override/) but it does not suggests that `amplify override api` is an alternative, as you ask me about all resources it I went a head and try `amplify override api` with all my API resources. 

**In the end, it seems like running override over all your resources it is the way to avoid this error.**

Thanks a lot for your response @akshbhu as it give me an idea of what to do. Wonder if there is a doc I am missing in terms of understanding the role of this override files and how them even when empty affect the build of the app.

**Comment by edwint88** [comment]
I have the same issue (also locally I run on macos), I've installed the amplify cli with curl and the amplify cli version is 7.4.5
My error:
`2021-12-03T18:00:35.318Z [INFO]: 🛑 Build error : ENOENT: no such file or directory, open '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json'
2021-12-03T18:00:35.318Z [WARNING]: ✔ Successfully pulled backend environment dev from the cloud.
2021-12-03T18:00:35.374Z [INFO]: 🛑 Build error : ENOENT: no such file or directory, open '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json'
2021-12-03T18:00:35.378Z [WARNING]: 2021-12-03T18:00:35.377Z - error: uncaughtException: Error: ENOENT: no such file or directory, open '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json' date=Fri Dec 03 2021 18:00:35 GMT+0000 (Coordinated Universal Time), pid=2170, uid=0, gid=0, cwd=/codebuild/output/src590624283/src/ng-website, execPath=/root/.nvm/versions/node/v12.21.0/bin/node, version=v12.21.0, argv=[/root/.nvm/versions/node/v12.21.0/bin/node, /root/.nvm/versions/node/v12.21.0/bin/amplify, pull, --amplify, {"envName":"dev","appId":"----removed-from-this-----"}, --providers, {"awscloudformation":{"configLevel":"project","useProfile":true,"profileName":"default"}}, --no-override, --no-codegen, --yes], rss=572575744, heapTotal=502734848, heapUsed=475964536, external=12890570, arrayBuffers=10837254, loadavg=[2.47, 1.32, 0.55], uptime=470, trace=[column=13, file=/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts, function=null, line=106, method=null, native=false, column=21, file=/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts, function=AmplifyAuthTransform.applyOverride, line=104, method=applyOverride, native=false, column=5, file=/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts, function=AmplifyAuthTransform.transform, line=72, method=transform, native=false, column=10, file=/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-stack-template.ts, function=generateAuthStackTemplate, line=6, method=null, native=false], stack=[Error: Error: ENOENT: no such file or directory, open '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json',     at /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts:106:13,     at AmplifyAuthTransform.applyOverride (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts:104:21),     at AmplifyAuthTransform.transform (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts:72:5),     at generateAuthStackTemplate (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-stack-template.ts:6:10)]`

I've tried the migration steps, it brings nothing to me (plus the override.ts is always empty)
Any suggestions?

**Comment by akshbhu** [comment]
Hi @edwint88 

can you try with latest amplify version 7.6.2 ?
If this issue still present , feel free to comment on this and ill open this issue again

**Comment by edwint88** [comment]
I suppose you've seen this https://github.com/aws-amplify/amplify-cli/issues/9209 - so you can close this one
